### PR TITLE
Add bitfinity support 

### PR DESCRIPTION
**Pull request by itsyaasir**
…ain support. This patch introduces the Ethereum JSON-RPC client dependency, updates Dockerfile for consistency, and adds support for the Bitfinity chain in the CLI.

### CODE REVIEW

File: bin/reth/src/utils.rs [hunk 1]
```diff
@@ -176,3 +185,86 @@ impl ListFilter {
...
+
+/// Bitfinity Genesis
+pub async fn bitfinity_genesis(url: String) -> Result<Arc<ChainSpec>> {
+    let client = ethereum_json_rpc_client::EthJsonRcpClient::new(ReqwestClient::new(url));
+
+    let chain_id = client.get_chain_id().await.map_err(|e| eyre::eyre!(e))?;
+
+    tracing::info!("Bitfinity chain id: {}", chain_id);
+
+    let genesis_block = client
```
Comment (ufoscout) [inline bin/reth/src/utils.rs:197]: This is not going to work if the blockchain does not have block 0 anymore.
The genesis block and genesis balances should be fetched lazily only if the importer needs them

File: crates/node-core/src/args/utils.rs [hunk 2]
```diff
@@ -77,6 +78,8 @@ pub fn genesis_value_parser(s: &str) -> eyre::Result<Arc<ChainSpec>, eyre::Error
         "holesky" => HOLESKY.clone(),
         #[cfg(not(feature = "optimism"))]
         "dev" => DEV.clone(),
+        #[cfg(not(feature = "optimism"))]
```
Comment (ufoscout) [inline crates/node-core/src/args/utils.rs:81]: why?

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: crates/node-core/src/args/utils.rs (hunk 2 above)
Comment (itsyaasir) [reply crates/node-core/src/args/utils.rs:81]: The bitfinity is only supported in non-optimism feature; the reth node can run on optimism(rollup node)

**Comment by ufoscout** [comment]
Fixed by #14

**Pull request closed without merging**